File: 442. Find All Duplicates in an Array/442. Find All Duplicates in an Array.py
class Solution(object):
    def findDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        nums.sort()
        B = []
        n = len(nums)
        for i in xrange(1, n):
            if nums[i] == nums[i - 1]:
                B.append(nums[i])
        return B

File: 442. Find All Duplicates in an Array/442. Find All Duplicates in an Array.java
public class Solution {
    public List<Integer> findDuplicates(int[] nums) {
        Arrays.sort(nums);
        List<Integer> repnums = new ArrayList<Integer>();
        int l = nums.length - 1;
        for (int i = 0; i < l; i++) {
            if (nums[i] == nums[i + 1]) {
                repnums.add(nums[i]);
            }
        }
        return repnums;
    }
}
